students/uss_tomasz/lesson_12_debugging/12_3_valdated_user_base/validate_user_base.py

```diff
@@ -0,0 +1,120 @@
+import argparse
+import csv
+import logging
+import os
+import re
+import sys
+
+
+EMAIL_REGEX = re.compile(r'''(
+    [a-zA-Z0-9._%+-]+  
+    @
+    [a-zA-Z0-9.-]+  
+    .
+    \.[a-zA-Z]{2,4}
+    )''', re.VERBOSE)
+PASSWORD_REGEX = re.compile(r'''
+        (?=.*[A-Z]) # positive look ahead assertions
+        (?=.*[0-9])
+        (?=.*[a-z])
+        .{8}
+        ''', re.VERBOSE)
+TELEPHONE_REGEX = re.compile(r'''(
+    (\+\d{2}\s)?
+    (
+    \d{3}
+    [\s-]?
+    \d{3}
+    [\s-]?
+    \d{3}
+    )
+    )''', re.VERBOSE)
+ZIP_CODE_REGEX = re.compile(r'\d\d-\d\d\d')
+REGEX_DATABASE = {'email': EMAIL_REGEX, 'password': PASSWORD_REGEX, 'telephone': TELEPHONE_REGEX,
+                  'zip_code': ZIP_CODE_REGEX}
+
+
+def parse_args(args):
+    parser = argparse.ArgumentParser('Collect database of mail, pass, tel and zip inside csv.')
```

Comment:
The parser can be global, since function arguments don't influence this object at all.
In this case it's not that important anyway, since this function is called once per script execution.